### DISCUSSION TITLE
Allow list for sanitised errors

### DISCUSSION THREAD

**Discussion by thomasverleye** [remix-run/react-router#12623] [Proposals]
All errors that are thrown to the client get sanitised to prevent leaking sensitive data.

It would be great to have the ability to add a **configuration** to allow certain attributes in an `Error` object.

The use-case for having this is because we want to share a correlation id and a certain error code from the original request that has happened (if available).

Now we're throwing this data with a status 500 and have to manually throw it to our error tracking (Sentry).

**Comment by manzaloros** [comment]
[Started a conversation about this here](https://github.com/remix-run/react-router/discussions/13213)

**Comment by jdevcc** [comment]
This would solve my issue #13314
